### ISSUE TITLE
objectgroup rotations occuring around incorrect point

### ISSUE THREAD

**Issue by davidjtferguson** [karai17/Simple-Tiled-Implementation#201]
In a game I'm working on we're using object groups to decorate our scenes but have come across the following indiscrepancy when rotating objects within an object group:

![rotation_discrepancy](https://user-images.githubusercontent.com/16323752/53582907-6057ec80-3b78-11e9-95ea-5f741f5321aa.png)

In the image you can see that in Tiled the object is rotated around the bottom left corner, whereas when it's loaded through STI it's rotated aproximately around the top right corner (although not exactly...)

On some investigation, it seems the issue is within `setObjectSpriteBatches(layer)` in init.lua, where the sprite batches for rotated tiles and objects is set up. There's a piece of code that looks like this:
```
			-- Compensation for scale/rotation shift
			if tile.sx == 1 and tile.sy == 1 then
				if tileR ~= 0 then
					tileY = tileY + tileH
					oy    = tileH
				end
			else
				if tile.sx < 0 then tileX = tileX + tileW end
				if tile.sy < 0 then tileY = tileY + tileH end
				if tileR   > 0 then tileX = tileX + tileW end
				if tileR   < 0 then tileY = tileY + tileH end
			end
```
which looks like it aims to fix the problem in question by moving the tileY down it's own size, then setting the origin to it's height, which means it'll draw in the original position but with the origin in the bottom left corner instead of the upper left corner meaning it'll rotate around that position (correctly). However that's set up for the tile size and not the object size (which explains why in the picture it's not quite the top left point that the rotated version is pivoted around).

So, I tried changing the code to the following just to experiment (keeping in mind this would probably break things for the tile rotation situation):
```
			-- Compensation for scale/rotation shift
			if tile.sx == 1 and tile.sy == 1 then
				if tileR ~= 0 then
					tileY = tileY + object.height
                                        oy    = object.height
				end
			else
				if tile.sx < 0 then tileX = tileX + object.width end
				if tile.sy < 0 then tileY = tileY + object.height end
				if tileR   > 0 then tileX = tileX + object.width end
                                if tileR   < 0 then tileY = tileY + object.height end
			end
```

However the results were... uuhh...

![object height](https://user-images.githubusercontent.com/16323752/53583641-e58fd100-3b79-11e9-8372-2f02b1cbee13.PNG)

... Not correct.

I'm going to keep investigating but thought I'd raise an issue to see if anyone has any ideas or if there's something obvious I'm missing.

**Comment by karai17** [comment]
The issue is likely that the offsets are wrong. LOVE's default pivot point
is the top-left corner. I believe Tiled's is the bottom-left corner. As you
can see in the code you posted, I do make a point to shange the pivot point
in LOVE by setting the "oy" (offset on y axis) variable.

You probably want to change the code in the object block, not the tile
block.

---
Landon Manning
lmanning17@gmail.com

**Comment by karai17** [comment]
If you have Discord, PM me there. It would be easier to figure out what is going on in a faster chat.

@Karai17#3209

**Comment by davidjtferguson** [comment]
Nice dude, thanks for getting back to me! I didn't realize there was an object block as well, I'll have a look for it. I believe I've sent you a Discord friend request, though i had to make the k in your tag capital.

**Comment by davidjtferguson** [comment]
This should be fixed as of PR #203
